test/Frontend/load-module-with-alias.swift

```diff
@@ -0,0 +1,20 @@
+/// Test the -module-alias flag.
+
+// RUN: %empty-directory(%t)
+
+/// Create a module Bar
+// RUN: echo 'public func bar() {}' > %t/FileBar.swift
+// RUN: %target-swift-frontend -module-name Bar %t/FileBar.swift -emit-module -emit-module-path %t/Bar.swiftmodule
+
+/// Check if Bar.swiftmodule is created
+// RUN: test -f %t/Bar.swiftmodule
+// RUN: not test -f %t/Cat.swiftmodule
+
+/// Create a module Foo that imports Cat with -module-alias Cat=Bar
+// RUN: echo 'import Cat' > %t/FileFoo.swift
+// RUN: %target-swift-frontend -module-name Foo -module-alias Cat=Bar %t/FileFoo.swift -emit-module -emit-module-path %t/Foo.swiftmodule -I %t
```

Comment:
Would it also be worth having a test for when the module "Cat" does exist on disk, but the module alias means another module gets loaded?

Reply:
yup will add the test